src/index.js

```diff
@@ -157,204 +125,173 @@ class IpfsRepo {
     return this.options.lock
   }
 
+  /**
+   * Opens the root backend, catching and ignoring an 'Already open' error
+   * @returns {Promise}
+   */
+  async _openRoot () {
+    try {
+      await this.root.open()
+    } catch (err) {
+      if (err.message !== 'Already open') {
```

Comment:
That would be good.  TBH I'm not sure where this message originates from - the default root datastore implementation is `datastore-fs` which does not throw an error with the message 'Already Open'.

There are [these standardised errors](https://github.com/ipfs/interface-datastore/blob/master/src/errors.js), maybe the best thing to do is to open a PR that expands that list with more error types?

Reply:
Thinking a little more about it, it seems like we often throw an error when reopening an already open thing, or reclosing an already closed thing, and then we have to write code like this with guards for that same error. It may make things simpler if we just ignore a double-open or double-close in our libs

Reply:
Sounds sensible, though is outside the scope of an callback->async/await refactor

Reply:
Fair enough 👍